parlai/mturk/tasks/acute_eval/run.py

```diff
@@ -0,0 +1,413 @@
+        default='I would prefer to talk to <Speaker 1>',
+        help='text next to speaker 1 radio button',
+    )
+    argparser.add_argument(
+        '--s2-choice',
+        type=str,
+        default='I would prefer to talk to <Speaker 2>',
+        help='text next to speaker 2 radio button',
+    )
+    argparser.add_argument(
+        '--question',
+        type=str,
+        default='Who would you prefer to talk to for a long conversation?',
+        help='question to present to turker for comparison (e.g. "Which speaker is better?")',
+    )
+    argparser.add_argument(
+        '--block-on-onboarding-fail',
+        type=bool,
+        default=True,
+        help='whether to block on onboarding failure',
+    )
+    argparser.add_argument(
+        '--subtasks-per-hit',
+        type=int,
+        default=5,
+        help='number of subtasks/comparisons to do per hit',
+    )
+    argparser.add_argument(
+        '--onboarding-threshold',
+        type=float,
+        default=0.75,
+        help='minimum accuracy on onboarding tasks, as a float 0-1.0',
+    )
+    argparser.add_argument('--seed', type=int, default=42, help='np.random seed')
+    argparser.set_defaults(allowed_conversation=1)
+    if from_argv:
+        return argparser.parse_args()
+    else:
+        return argparser.parse_args(args=[])
+
+
+def setup_task_queue(opt):
+    """
+    Initialize task queue to contain the specified number of instances of each pairing.
+    """
+    # hacky fix for the parlai parser hacky fix
+    annotations_per_pair = opt['annotations_per_pair']
+    internal_pair_id = 0
+
+    ## Set up onboarding tasks
+    if opt['task_folder']:
+        filename = os.path.join(opt['task_folder'], opt['pairings_filename'])
+        with open(filename, 'r') as pf:
+            for l in pf:
+                pairing_dict = json.loads(l.strip())
+                read_task_from_jsonl(
+                    pairing_dict,
+                    internal_pair_id,
+                    opt['s1_choice'],
+                    opt['s2_choice'],
+                    opt['question'],
+                )
+                internal_pair_id += 1
+    else:
+        raise Exception("You must provide a task folder")
+
+    ## Fill task queue
+    for _i in range(annotations_per_pair):
+        all_task_keys = list(desired_tasks.keys())
+        np.random.shuffle(all_task_keys)
+        for internal_pair_id in all_task_keys:
+            task_queue.put(desired_tasks[internal_pair_id])
+    # limit number of hits worker can do by default
+    if opt['max_hits_per_worker'] == 0:
+        opt['max_hits_per_worker'] = (
+            len(desired_tasks) + len(onboarding_tasks)
+        ) // opt['subtasks_per_hit']
+
+
+def read_task_from_jsonl(
+    pairing_dict, internal_pair_id, s1_choice, s2_choice, question
+):
+    """
+    Build task dict according to expected format.
+    """
+    conv_orders = [[0, 1], [1, 0]]
+    conv_order = conv_orders[np.random.choice([0, 1])]
```

Comment:
nit: can get rid of these two lines and just do `conv_order = random.choice([[0, 1], [1, 0]])`

Reply:
alternatively, it seems you only really need a bool to track this conversation order (i.e. you only care about the first index)